src/core/midi/MidiClient.cpp

```diff
@@ -221,10 +221,12 @@ void MidiClientRaw::parseData( const unsigned char c )
 		case MidiNoteOff:
 		case MidiNoteOn:
 		case MidiKeyPressure:
+			// set velocity of key
+			m_midiParseData.m_midiEvent.setVelocity(m_midiParseData.m_buffer[1]);
```

Comment:
There really should be an explicit comment or annotation when falling through in a `switch`-`case`.